test/trip_dispatch_test.rb

```diff
@@ -122,4 +122,52 @@ def build_test_dispatcher
       end
     end
   end
+
+  describe "request a new trip" do
+    before do
+      @dispatcher = build_test_dispatcher
+      @new_trip = @dispatcher.request_trip(2)
+    end
+
+    it "the new trip must be a Trip class" do
+      expect(@new_trip).must_be_instance_of RideShare::Trip
+    end
+
+    it "update driver's status to unavailable" do
+      expect(@new_trip.driver.status).must_equal :UNAVAILABLE
+    end
+
+    it "update passenger's trip list" do
+      passenger = @dispatcher.find_passenger(2)
+      expect(passenger.trips.last).must_equal @new_trip
+    end
+
+    it "update driver's trip list" do
+      driver = @new_trip.driver
+      expect(driver.trips.last).must_equal @new_trip
+    end
+
+    it "raise error when no driver is available" do
+      @dispatcher.driver.each do |driver|
+        driver.status = :UNAVAILABLE
+      end
+      expect { @dispatcher.request_trip(9) }.must_raise ArgumentError
+    end
+
+    it "select a driver whose has no trip" do
+      expect(@new_trip.driver_id).must_equal 3
+    end
+
+    it "select a driver who has the earliest end time" do
```

Comment:
You should also have a test where it selects a driver with no trips, if that driver exists.